src/main/webapp/script.js

```diff
@@ -160,6 +160,57 @@ async function processImage(image) {
   drawImageOnCanvas(blurredImage.object, outputCanvas);
 
   updateDownloadButton(blurredImage);
+
+  outputCanvas.onclick = (event) => {
+    tryToUnblurArea(event, image);
+  };
+}
+
+/**
+ * Function to handle click on output canvas
+ * and toggle blurness of rects to blur
+ * if any were clicked.
+ * @param {MouseEvent} event
+ * @param {ImageObject} image
+ */
+function tryToUnblurArea(event, image) {
+  const outputCanvas = document.getElementById('output-canvas');
+
+  // coordinates of click.
+  const eventX = event.offsetX;
+  const eventY = event.offsetY;
+
+  for (const blurRect of image.blurAreas) {
+    const scaleForCanvas = (propertyToScale) => {
+      return propertyToScale * outputCanvas.clientWidth / image.object.width;
+    };
+
+    // blurRect has all properties in original image coordinates,
+    // click event is in canvas coordinates.
+    // Scale blurRect to be in canvas coordinates.
+    const scaledRect = {
+      'leftX': scaleForCanvas(blurRect.leftX),
+      'topY': scaleForCanvas(blurRect.topY),
+      'rightX': scaleForCanvas(blurRect.leftX) +
+          scaleForCanvas(blurRect.width),
+      'bottomY': scaleForCanvas(blurRect.topY) +
+          scaleForCanvas(blurRect.height),
+    };
+
+    // if we clicked on this rect - toggle its visibility.
+    if (scaledRect.leftX <= eventX && eventX <= scaledRect.rightX &&
+        scaledRect.topY <= eventY && eventY <= scaledRect.bottomY) {
+      blurRect.toBeBlurred = !blurRect.toBeBlurred;
+
+      const blurRadiusInput = document.getElementById('blurring-radius');
+
+      const blurredImage = getImageWithBlurredAreas(
+          image, blurRadiusInput.value);
+      drawImageOnCanvas(blurredImage.object, outputCanvas);
+
+      updateDownloadButton(blurredImage);
```

Comment:
I think lines 205-211 were also used somewhere else. Isn't it better if you create a function for them, for example, updateBlurredImage(image)?

Reply:
Also, I think it's better to move these lines (or the function) out of the for, and call them if the user clicked at least one rectangle. There is a small chance of a user clicking 2 rectangles at once (if they overlap) and reblurring the whole image 2 times is not needed.

Reply:
Yes, thanks!